### PR TITLE
Dockerfile: switch to bullseye

### PR DESCRIPTION
**Pull request by kolyshkin**
For one thing, this should help PR #2862 (see https://github.com/opencontainers/runc/pull/2682/commits/a0e755efd86b5e474e4874c6c25ceb16bf4af30e#r684182923)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ARG GO_VERSION=1.17
 ARG BATS_VERSION=v1.3.0
 
-FROM golang:${GO_VERSION}-buster
+FROM golang:${GO_VERSION}-bullseye
 ARG DEBIAN_FRONTEND=noninteractive
 
 RUN echo 'deb https://download.opensuse.org/repositories/devel:/tools:/criu/Debian_10/ /' > /etc/apt/sources.list.d/criu.list \
```
Comment (AkihiroSuda) [inline Dockerfile:7]: I guess this line needs to be updated too, but I don't see `Debian_11` repo there, so we can update later.

**Review by AkihiroSuda — APPROVED** [review]

**Review by rata — APPROVED** [review]
LGTM. I'll rebase #2682 once this is merged

**Comment by rata** [comment]
Actually, I just cherry-picked it there :-)

File: Dockerfile (hunk 1 above)
Comment (kolyshkin) [reply Dockerfile:7]: Yes, I took a look, there's no Debian_11 repo and the existing one works fine.

@adrianreber hope you can add it later (either at the time of 3.16 release, or earlier).

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**